201-vm-customimage-new-storageaccount/azuredeploy.json

```diff
@@ -0,0 +1,307 @@
+﻿{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "TransferVmName": {
+      "type": "string",
+      "defaultValue": "TransferVM",
+      "minLength": 3,
+      "maxLength": 15,
+      "metadata": {
+        "description": "Name of the Windows VM that will perform the copy of the VHD from a source storage account to the new storage account created in the new deployement, this is known as transfer vm."
+      }
+    },
+    "NewVmName": {
+      "type": "string",
+      "defaultValue": "NewVM",
+      "minLength": 3,
+      "maxLength": 15,
+      "metadata": {
+        "description": "Name of the new VM deployed from the custom image."
+      }
+    },
+    "CustomImageName": {
+      "type": "string",
+      "defaultValue": "images/Win10MasterImage-osDisk.72451a98-4c26-4375-90c5-0a940dd56bab.vhd",
+      "metadata": {
+        "description": "Name of the VHD to be used as source syspreped/generalized image to deploy the VM. E.g. mybaseimage.vhd."
+      }
+    },
+    "AdminUserName": {
+      "type": "string",
+      "defaultValue": "vadmin",
+      "minLength": 1,
+      "metadata": {
+        "description": "Name of the local administrator account, this cannot be Admin or Administrator or root if Linux VM."
+      }
+    },
+    "AdminPassword": {
+      "type": "securestring",
+      "minLength": 8,
+      "metadata": {
+        "description": "Local administrator password, complex password is required, do not use any variation of the password word because it will be rejected."
+      }
+    },
+    "sourceStorageAccountKey": {
+      "type": "string",
+      "metadata": {
+        "description": "Key of the storage account that contains the custom image."
```

Comment:
If you know the source storage account name and resource group it's better to use the listKeys function to find the key in template execution than ask the user to submit a key